### ISSUE TITLE
💡 Discussion on vocabulary for data

### ISSUE THREAD

**Issue by RoteKekse** [nfdi4cat/voc4cat#45]
### Description

Dear NFDI4Cat,

I am working at HZB for Catalysis and I am using Nomad as a software/ELN. We use nomad to describe catalytic data.

I met David Linke Last Wednesday and I would like to discuss if it is worth adding terms describing the data.

In an Ontology there is usually the distinction between a `specific dependent quality` for example `selectivity (of a product)` and a data point for example `50%`. Selectivity is a property a catalyst always has when probed in a specific catalytic experiment with specific conditions. Data also usually as the property of a unit or dimensionless.
The data point(s) could be the output of a measurement, they could be the results of a simulation or of a calculation. And there can be multiple of those. E.g. the experiment could be conducted under the same conditions multiple times on different days.+

For selectivity one could argue that any catalyst has a selectivity for any possible product (although it might be 0 quite a lot). And the fact that it has that does not depend on any data it is just something it has.

Data then reveals how this quality could be quantified in a specific situation.

This distinction is  also present in the ontology work such as in the OBI.

I find this distinction quite helpful since it separates the data from the quality. A more practical distinction would be the mass of a human. Any human has a mass. The fact that they have a mass is fixed and a quality of every human. That mass then can be measured and data is produced, could be 80kg, 100000g or others. 

For me as a data steward this makes life a lot easier and i applied this already in the context of solar cells.

What do you think :)?

@schumannj, @dalito @HendrikBorgelt @AleSteB 

All the best
Micha

**Comment by HendrikBorgelt** [comment]
Hi Micha, 

while I accept that selectivity is a property that every catalyst has and that can be uniquely attributed to a catalyst, using selectivity in the same sense as weight is quite difficult. We might forgo that the weight must be measured at a certain time since it is mostly unimportant whether a person is around 100kg today and was 99,8kg yesterday, it is still around 100kg. We can not do so with the selectivity. Selectivity is a nonunary characteristic. It always depends on the given educts and the respective product to which it refers. So "selectivity(of a product for a given educt)" Measuring the Selectivity of a deNOx catalyst like a car catalyst also depends on the fuel we use, not only whether we measure NO2 or N2O2. And even if we can say this, the selectivity is still dependent on the whole set of reaction conditions (e.g. Cold start selectivity vs set point selectivity). If you are familiar with I-adopt this picture, where I tried to model the data content of https://nfdirepo.fokus.fraunhofer.de/dataset.xhtml?persistentId=doi:10.82207/FK2/NR5BWO via I-adopt, might help you understand the complexity of modeling a selectivity measurement in a simple data set.
![Blank diagram - Page 1](https://github.com/nfdi4cat/voc4cat/assets/84382772/9e76921d-424c-47e5-8733-c81ef186125f)

Maybe it is best not to focus on a highly composite term like selectivity (it is basically not one but multiple parameters combined to be more easily comparable) but focus more on more straightforward terms like reaction enthalpy or even decompose a term like selectivity into more manageable terms like conversion and yield.

But I would also like to hear your suggestion as to how you either want to handle the complexity of a selectivity or examples in which you want to use the term selectivity without struggling with such things.

all the best
Hendrik

**Comment by RoteKekse** [comment]
Hey Hendrik,

thanks for your answer :). I think we need to boil it down to the point where it is like a mass of a person. The data point is the output of a measurement and therefore the context like time, reactants, temperature profile etc is given. If we have a term like `selectivity measurement datum` this then can have in an ontology sense "children" such as `selectivity of methane measurement datum`. The instances of such a thing are then the specific values of a specific measurement. By the way for mass of a human it is not tha much different, it also depends on temperature, time, air pressure and much more, it just feels easy because we now of what it is a quality. The question remains of what is selectivity a quality. Below I would say that selectivity/selectivities are a quality of a chemical reaction.  

Lets be concrete, since you asked for an example, i will use highlighted notation to indicate triple stores:
Lets say we have a `catalytic measurement`. The  `catalytic measurement` `has specific inputs` `temperature setting datum`, `pressure setting datum`... which `are` `data points` and `are specifications of` `process temperature`, `process pressure` ... Note that the data points could easily be a profile over time.  The `catalytic measurement` `contains` `catalytic reaction`. The `catalytic reaction` `is a` `chemical reaction` and  `has reactant` some `chemicals` eg `CO` and `has products` some `chemicals` eg `Methane` and `has catalyst` some `catalyst`.
Then `conversion` `is a qualitiy of` each `reactant`. While the `conversion measurement datum` `is output of` the `catalytic measurement`.  
Selectivity is a bit more tricky. Since it is a quality of a ration of reactants and products. So `selectivity` `is a qualitiy of`   `chemical reaction`. While the `selectivity measurement datum` `is output of the whole `catalytic measurement`.

This vocabulary could be used quite nicely to annotate data in nomad. I still find it quite helpful to separate the qualities like mass from the data. Because then you can think about of what mass is a quality and then at another point you can think about where the data comes out, eg a measurement. 

Obviously one could use one term for data and quality, but then i always wonder, are you referring to selectivity as a property of an reaction or are you referring to a numerical value of a specific measurement.

What do you think, best Micha

**Comment by schumannj** [comment]
Yes, thanks David, I copied my last paragraph into the new issue  #46.

**Comment by HendrikBorgelt** [comment]
Hi @schumannj and @RoteKekse and all others,

sorry if I misunderstood the `selectivity (of a product)`, I think I was just interpreting it badly. I agree that we should have something like a `selectivity measurement datum`, however, I would still recommend against a direct `reaction selectivity measurement datum` as to my knowledge selectivity in a chemical sense is never directly measured but always computed (correct me if I am wrong and don't know the right measurement technique). This also eliminates the question of whether an entity has a selectivity or not and therefore reduces the complexity of describing the selectivity. I know that most "measurements" in the field of chemistry directly reduce their data into something like a selectivity, however, I don't know if this should be best practice and if we thus should model it in such a way, when describing data. Unfortunately, I have to admit that I work relatively little on real data in the context of selectivity determinations. I would therefore recommend that you simply send me another comment as to whether my suggestion makes sense or not.

With regards to @schumannj, I think we should define all those measurement datum, as the class "conversion" for example might just be used in a Thesauri sense while the class "conversion measurement datum" is the only one allowed to inherit Individuals which link to the data of the measurement. The question for me is more where the pure application level begins and where our direct ontology ends. I recommend reading this [design pattern](https://nfdi4chem.github.io/VibrationalSpectroscopyOntology/design_patterns/) and the corresponding [example](https://github.com/NFDI4Chem/VibrationalSpectroscopyOntology/blob/main/src/ontology/components/vibso_examples.ttl), maybe that will help. As you can see in the links, the 4Chem data would just be modeled.

regarding the difficulties in finding a definition for selectivity, I agree.

**Comment by dalito** [comment]
Even after the split there may still be two discussion topics here:

- How to model catalysis data?
- Which terms should voc4cat provide?

Regarding the latter:
- First, voc4cat should provide at least all definitions for the terms used in a datamodel. Most important is that voc4cat has at least all terms for a normalized or narrow data model in RDMS world or TBox in ontology world.
- Second, voc4cat may provide more definitions/terms which are not strictly necessary for a data model but are established terms in the (sub-)domains. For example "catalytic reactor" is kind of redundant, if reactor and catalytic reaction are defined.
- Third, voc4cat may provide collections of terms for various purposes (in applications), e.g. we could make a catalyst performance measure collection or a synthesis method collection etc.. Note that one term can be part of multiple collections and that collections can be part of other collections in SKOS.

On "datum" in the previous messages: You seem to talk about something different than a location "a datum is a reference point, surface, or axis on an object against which measurements are made" (https://en.wikipedia.org/wiki/Datum_reference). What do you mean with datum?

**Comment by HendrikBorgelt** [comment]
In the OBI and IAO, to which @RoteKekse and @schumannj or at least I am referring, a datum is just the singular of a [data item](https://www.ebi.ac.uk/ols4/ontologies/iao/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FIAO_0000109). 
So I hope that we all speak of the same idea of datum as also found in the [Wikipedia article of data](https://en.wikipedia.org/wiki/Data): 

> In common [usage](https://en.wikipedia.org/wiki/Usage_(language)) data ([US](https://en.wikipedia.org/wiki/American_English): [/ˈdætə/](https://en.wikipedia.org/wiki/Help:IPA/English); [UK](https://en.wikipedia.org/wiki/British_English): [/ˈdeɪtə/](https://en.wikipedia.org/wiki/Help:IPA/English)) is a collection of discrete or continuous [values](https://en.wikipedia.org/wiki/Value_(semiotics)) that convey [information](https://en.wikipedia.org/wiki/Information), describing the [quantity](https://en.wikipedia.org/wiki/Quantity), [quality](https://en.wikipedia.org/wiki/Qualitative_property), [fact](https://en.wikipedia.org/wiki/Fact), [statistics](https://en.wikipedia.org/wiki/Statistics), other basic units of meaning, or simply sequences of [symbols](https://en.wikipedia.org/wiki/Symbol) that may be further [interpreted formally](https://en.wikipedia.org/wiki/Interpretation_(logic)). A datum is an individual value in a collection of data.

**Comment by HendrikBorgelt** [comment]
Regarding the question whether we need to further split the discussion, I think splitting is not necessary as I think both question can not be answered mutually exclusive.

If we want to provide an OBI/IAO-esque approach to providing data we are "stuck" with the general terms and ideas provided via both Ontologies. If we do not want to align your vocabulary (at least too much) to those ontologies we might be able to answer both questions separately.

**Comment by RoteKekse** [comment]
Hey Hendrik, 
In the OBI world there is the distinction between a datum and a measurement datum. The first is the output of a data transformation while the second is the output of an assay. I am totally fine with selectivity data being not measured but an output of a Data transformation. Nevertheless selectivity can be a quality of a reaction.

The same is true with efficiency of a solarcell. It is calculated from a measurement but a quality of a solar cell.

I would be happy if we could include these distinctions in voc4cat :). I find the initiative very helpful and was happy to learn it last week. I think this can be a big step towards interoperability in catalysis! Me and Julia could try out to annotate data then in nomad to test the vocabulary in an ELN setting.

best Micha

**Comment by dalito** [comment]
I agree that selectivity is most often an output of a data transformation. However, you can get devices that output selectivity data directly and do the transformation internally. So selectivity is not fundamentally different then other measures. Temperature is also the result of a transformation, e.g. from voltage to Kelvin. The difference is that in one case you trust the reading being a result of an "external" transformation and in the other case you include the transformation in the data model. Both cases are relevant.

**Comment by RoteKekse** [comment]
Hey David, you are correct, for us it is important to know where to draw a line for now (we can extend it if needed). We do not need to describe reality as its fullest (which is any how not possible :D) and we need to compromise somewhere. For catalysis research i think it is fine group temperature as a measurement (might not be if you are a control engineer) for the selectivity that might be different. But if not we can just treat selectivity as the measurement datum for now and if needed we can refine later. 

This process is never finished and we have always the option to refine if needed.

**Comment by dalito** [comment]
Yes, be pragmatic first. ...but also be aware of the simplifications you made. I am looking forward to hear how well the nomad ELN integration goes.

**Comment by dalito** [comment]
@RoteKekse will you closely follow OBO-style modelling? I wonder because you were referring to their ontologies mainly.

@HendrikBorgelt is leaning towards I-ADOPT and DCAT. I-ADOPT should interface well with voc4cat as terminology.

It would be great to have an exchange about how well both approaches work. Maybe at or soon after Katalytiker-Tagung?

**Comment by dalito** [comment]
Regarding the modelling of selectivity, which is a ratio of two or more measurements, you may find the example for [modelling body-mass index (BMI) in OBI](https://ddooley.github.io/docs/data-bmi/#) interesting. BMI is similar in the way that it is also derived from two measurements.

**Comment by HendrikBorgelt** [comment]
@dalito, regarding I-ADOPT and DCAT, i am currently just trying to get something in between a data format with is unstructured but uses a defined vocabulary such as "Voc4Cat" and "an" Ontology (in Catalysis I think we mostly have to deal with a plethora of Ontologies combine into the "local" Ontologie), which I think is too difficult for most Scientists and even Data Stewards to handle. So my focus is more on, how can we use a "Vocabulary" such as DCAT to give Scientists as well as Datastewards, an easy model after which they can structure their data. (something like a conceptual model which might not be perfectly depicted in this picture https://www.researchgate.net/publication/234793155/figure/fig1/AS:595782646378496@1519057060120/More-Expressive-Semantic-Models-Enable-More-Complex-Applications.png )

**Comment by HendrikBorgelt** [comment]
@RoteKekse I think the OBI/IAO approach is one of the most flexible and also expressive modeling descriptions one can have, it is just quite large and therefore "unwieldy" compared to an approach such as 
![grafik](https://github.com/nfdi4cat/voc4cat/assets/84382772/10fabba9-e1e5-4909-9bd8-72991b0035db)
from i [adopt](https://www.rd-alliance.org/group/interoperable-descriptions-observable-property-terminology-wg-i-adopt-wg/wiki/i-adopt). This is of course shorter but some times also causes troubles in not being flexible enough. See:
![Blank diagram - Page 1 (1)](https://github.com/nfdi4cat/voc4cat/assets/84382772/8fd82e3c-147e-42ff-92b3-76844b78cea9)
which tries to model a High throughput selectivity measurement of 4 different types of catalyst as can be found here: https://nfdirepo.fokus.fraunhofer.de/dataset.xhtml?persistentId=doi:10.82207/FK2/NR5BWO

**Comment by HendrikBorgelt** [comment]
@RoteKekse, I have by chance gone through the issues of voc4cat again and have seen that we have never closed this issue. Since we have a new development in progress, I wanted to ask if there is still interest in this issue. Just for your information, we are currently extending a DCAT model in the context of NFDI4Chem and 4Cat so that we can also model more complex content. I don't know if such a model is of interest.

If there is no further interest in this issue, I would just ask whether we should close it or not.

**Comment by RoteKekse** [comment]
Hey thanks for letting me know. will this then all be in the voc4cat or this a separate development?

**Comment by HendrikBorgelt** [comment]
its a separate approach, where we can reuse a lot of voc4cat but also other terms from ontologies a.s.o. If you are interessted, maybe contact me via mail, so i can present you our approach. Otherwise you can find most of the schema here: https://github.com/StroemPhi/dcat-4C-ap 

But be warned, it is written mostly in linkML (the other artefakts like json are derived from linkML) and since it is only the raw schema, it might be complicated without explenation due to beiing in development and some quirks of LinkML.

**Comment by dalito** [comment]
In general the issue goes into data modelling. voc4cat as SKOS vocabulary should be seen as terminology but not as tool for data modelling. For this we use linkml as proxy for SHACL, JSONschema, SQL models. One example demonstrating this combined approach is the pid4cat-model where voc4cat provides term/definitions for [enums](https://nfdi4cat.github.io/pid4cat-model/latest/elements/RelationType/#permissible-values).

@RoteKekse - Do you currently use voc4cat in some real application? We want to add some real-life uses in the documentation under ["How to use?"](https://nfdi4cat.github.io/voc4cat/docs_usage/how-to-use.html).

I would also suggest to close the issue (after the reply).
